### PR TITLE
Copy name related code from Malaysia Person class to Singapore Person class

### PR DESCRIPTION
**Pull request by ziming**
Currently Singapore Person class has no name related methods, which mean that when you call

$this->faker->name()

you will get either Jane Doe or John Doe, this also cause issues when you do $this->faker->unique()->name()

since you only have 2 names in the default Person super class.

Now, Malaysia and Singapore are 2 neighbour countries that share many similarities when it come to person names. Residents of both countries often travel to each other for leisure and work reasons too. Making it a great class for the Singapore Person class to inherit from.

### CODE REVIEW

**Comment by pimjansen** [comment]
This does not feel right. Both belong to another language so please lets not mix in between. If you want proper names, update the pr to reflect it in the locale itself

**Comment by ziming** [comment]
so ur fine with me copying all the name related methods and variables over?

or can I hold an instance of the Malaysia person class in the singapore person class and proxy calls to it?

**Comment by pimjansen** [comment]
> so ur fine with me copying all the name related methods and variables over? which is pretty much everything except the malaysian Id method
> 
> or can I hold an instance of the Malaysia person class in the singapore person class and proxy calls to it?
> 
> let me know which u prefer and I update the pr

Copy it over since both are seperate languages. For Faker 2.x which is the next major all will consist of language packs so there cant be a mix between those

**Comment by ziming** [comment]
Alright, I have copied them over.

**Comment by pimjansen** [comment]
Please update the PR according the contribution guidelines. I see no tests at all etc

**Comment by ziming** [comment]
oh ok I will copy the tests over too later

**Comment by pimjansen** [comment]
> oh ok I will copy the tests over too later. am sick now

Fine, but also please review the test to see if they are valid. We will review this as new code

**Comment by ziming** [comment]
Hi

I just went to the ms_MY Person test file and found that it has no test for it's name methods

https://github.com/FakerPHP/Faker/blob/main/test/Faker/Provider/ms_MY/PersonTest.php

I then went to check a few other related locales since Singapore and Malaysia is largely made up of (Chinese, Malays, Indians, English).

English: (no name tests)
https://github.com/FakerPHP/Faker/blob/main/test/Faker/Provider/en_US/PersonTest.php

https://github.com/FakerPHP/Faker/blob/main/test/Faker/Provider/en_GB/PersonTest.php

Chinese: (no name tests)
https://github.com/FakerPHP/Faker/blob/main/test/Faker/Provider/zh_TW/PersonTest.php

Malay: (no name tests)
https://github.com/FakerPHP/Faker/blob/main/test/Faker/Provider/ms_MY/PersonTest.php

Indian: (no name tests)
https://github.com/FakerPHP/Faker/tree/main/test/Faker/Provider/en_IN

So this mean i have to try write my own tests for the names, which will take longer, hope you don't mind waiting until my covid is over. So it may take longer than this week.

**Comment by pimjansen** [comment]
No worries! Better to have solid code. Will pin the PR for the time being

**Comment by ziming** [comment]
Hi, added the tests.

**Comment by ziming** [comment]
Hmm I should have used assertIsString instead of assertNotEmpty for some of them

**Comment by ziming** [comment]
Hi @pimjansen I changed the assertion for those where the result might be empty string to assertIsString

**Comment by ziming** [comment]
Hi @pimjansen anything else I need to do?

Thank you

**Comment by pimjansen** [comment]
I did not look yet, currently kn leave but i will review once im back

**Comment by ziming** [comment]
hi @pimjansen hope u enjoyed ur leave. is this okay now?

**Comment by bram-pkg** [comment]
For some reason the pipeline is not running.

**Comment by ziming** [comment]
Anything I can do to help resolve that or is that something on the faker team side?

**Comment by bram-pkg** [comment]
You could try making an empty commit.

**Comment by ziming** [comment]
As you can see I tried to make an empty commit but the github actions still did not run

**Comment by DvDty** [comment]
Seems like the pipeline jobs are queued, but are "awaiting approval from a maintainer"?
Example: https://github.com/FakerPHP/Faker/actions/runs/4766832639

**Review by aznj — APPROVED** [review]

**Comment by bram-pkg** [comment]
image
I do not see an admin approval button.

Please fix the tests for PHP 7.4 and the other pipeline actions

**Comment by pimjansen** [comment]
> image
> 
> I do not see an admin approval button.
> 
> 
> 
> Please fix the tests for PHP 7.4 and the other pipeline actions

I approved whehe

File: test/Faker/Provider/en_SG/PersonTest.php
Comment (pimjansen) [inline test/Faker/Provider/en_SG/PersonTest.php:15]: They will never be empty due to how interfacing works. Please assert actual values with seeding instead.

**Review by pimjansen — CHANGES_REQUESTED** [review]
See my review changes on the tests

**Pull request closed without merging**
